### DISCUSSION TITLE
Component style of parent components of FluentCard

### DISCUSSION THREAD

**Discussion by martinbu** [microsoft/fluentui-blazor#1399] [General]
Hi,

I have noticed that if a FluentCard is used inside a FluentStack or FluentTab that a lot of css variables are attached to FluentStack/FluentTab. Is there a reason for this?

![grafik](https://github.com/microsoft/fluentui-blazor/assets/1791936/d77312b9-5df1-41f1-a35b-aab177d3c4b4)

The reason for this question is
1. Interest
2. I am overriding some css variables directly to be able to adjust my style to a style guide. If I am using the design tokens, the system always changes my colors to some degree.

**Comment by vnbaaij** [comment]
Hi Martin,

1) This is done by the underlying web components script. I believe it should actually be 'injected' in the css but not sure why that is not the case. Will ask the Fluent team
2) A design token color is always processed by a recipe to make sure the combination of the background it is shown on and the color itself have enough contrast between them (a11y requirement). Therfore the old you specify will almost never be exactly what is shown on screen

**Reply by martinbu** [reply]
Hi Vincent,

1. Thank you for that.
2. Yes I know that and already do some research about the "issue" of getting the colors that the designers want to have. I started with this [issue](https://github.com/microsoft/fluentui-blazor/issues/144#issuecomment-1033119655). Here the adaptiv colors and recipes are explained - which is a good thing, don't get me wrong. But in the post of bheston, he mention an "ideal" color recipe algorithm in #5470. I think it is this [pull](https://github.com/microsoft/fast/pull/5470) and the recipe is maybe [this one](https://github.com/microsoft/fast/pull/5470/files#diff-6dd51a261d283c1d8f74f1b91d0cdd212afb2058fa56034cd3f63d0a2393d044).
With this recipe I had the hope to get a color that more likly corresponds to the color defined in AccentBaseColor. But I couldn't find this "ideal" algorithm anywhere in the documentation nor in the code base. Not in fluentui-blazor and not in the fast code base.
As an example of the issue that I have with colors and why I set the accent-fill-rest variable to the target color with `--accent-fill-rest: #1f457a !important;`
If I set the AccentBaseColor to #1F457A
    1. On https://color.fast.design/ I get very nice and accurate colors
    ![grafik](https://github.com/microsoft/fluentui-blazor/assets/1791936/0c542468-c89e-4b56-abc2-ceec6b1e84d2)

    2. But with the "new system" in FluentUI Blazor I get this color for buttons, etc. which doesn't even come close to meet the defined color. The designers won't accept that.
    ![grafik](https://github.com/microsoft/fluentui-blazor/assets/1791936/7ad6617f-c088-4ad9-8c34-e3a8cfef79da)

Do you know anything about this ideal color recipe? And/Or do you see the need of such an algorithm too, like me and maybe others?

**Reply by vnbaaij** [reply]
As Brian explains in the linked issue, what is shown on color.fast.design is not representative of what is in the package now. Fot the merged PR with the ideal recipe, I see that branch still exists and has never been merged into the main branch.
@bheston can you explain what the latest state on the recipes is?

**Reply by bheston** [reply]
Nice detective work. Yes, I believe you found all the correct references. It looks like that "ideal" recipe was added to `@microsoft/fast-components`, which is the deprecated package. You could take that code if you want and update the accent recipe design token. This would be something like `accentFillRecipe.withDefault(...)`. Use [this line](https://github.com/microsoft/fluentui/blob/5aefa438776a47278b09f5073149bb94fb14c044/packages/web-components/src/design-tokens.ts#L425C14-L425C30) as an example, but instead of calling `contrastAndDelta....` two lines below, call the `ideal` function.

The updated color model was aligned with Windows 11 which adjusted the button colors for light and dark mode. An example I often used was the dark purple Microsoft brand color. This looked fine in light mode but not on a dark surface, so the algorithm adjusted that purple (in both modes) to maintain a similar contrast from the background. It really is a beneficial feature, but does cause some conflict with brand colors. It really is the only way (I believe) to handle the needs of brands and customizations in both light and dark mode while maintaining contrast without doing a bunch of manual tracking to make sure it never breaks.

Hope this helps.
